shared/src/commonMain/kotlin/co/nimblehq/avishek/phong/kmmic/data/remote/ApiClient.kt

```diff
@@ -0,0 +1,126 @@
+package co.nimblehq.avishek.phong.kmmic.data.remote
+
+import co.nimblehq.avishek.phong.kmmic.BuildKonfig
+import co.nimblehq.avishek.phong.kmmic.data.local.datasource.TokenLocalDataSource
+import co.nimblehq.avishek.phong.kmmic.data.remote.body.RefreshTokenApiBody
+import co.nimblehq.avishek.phong.kmmic.data.remote.datasource.TokenRemoteDataSource
+import co.nimblehq.avishek.phong.kmmic.domain.model.AppError
+import co.nimblehq.jsonapi.json.JsonApi
+import co.nimblehq.jsonapi.model.JsonApiException
+import io.github.aakira.napier.DebugAntilog
+import io.github.aakira.napier.LogLevel
+import io.github.aakira.napier.Napier
+import io.ktor.client.*
+import io.ktor.client.engine.*
+import io.ktor.client.plugins.auth.*
+import io.ktor.client.plugins.auth.providers.*
+import io.ktor.client.plugins.contentnegotiation.*
+import io.ktor.client.plugins.logging.*
+import io.ktor.client.plugins.logging.LogLevel.*
+import io.ktor.client.request.*
+import io.ktor.client.statement.*
+import io.ktor.http.*
+import io.ktor.serialization.kotlinx.json.*
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.flow.last
+import kotlinx.serialization.json.Json
+
+class ApiClient(
+    engine: HttpClientEngine,
+    tokenLocalDataSource: TokenLocalDataSource? = null,
+    tokenRemoteDataSource: TokenRemoteDataSource? = null
+) {
+
+    val httpClient: HttpClient
+    val json = Json {
+        prettyPrint = true
+        isLenient = true
+        encodeDefaults = true
+        ignoreUnknownKeys = true
+    }
+
+    init {
+        Napier.takeLogarithm()
+        Napier.base(DebugAntilog())
+
+        httpClient = HttpClient(engine) {
+            install(Logging) {
+                level = ALL
+                logger = object : Logger {
+                    override fun log(message: String) {
+                        Napier.log(LogLevel.DEBUG, message = message)
+                    }
+                }
+            }
+
+            install(ContentNegotiation) {
+                json(json)
+            }
+
+            if (tokenLocalDataSource != null && tokenRemoteDataSource != null) {
+                install(Auth) {
+                    bearer {
+                        loadTokens {
+                            tokenLocalDataSource.getToken()?.run {
+                                BearerTokens(accessToken, refreshToken)
+                            }
+                        }
+
+                        sendWithoutRequest { request ->
+                            request.url.host == Url(BuildKonfig.BASE_URL).host
+                        }
+
+                        refreshTokens {
+                            tokenRemoteDataSource
+                                .refreshToken(
```

Comment:
How about this to reduce an indentation level?
```suggestion
                            tokenRemoteDataSource.refreshToken(
```

Reply:
I think breaking the new line at `.refreshToken` will make it look consistent with `.last()` and `run` below.